### PR TITLE
Allow Passing custom no results component

### PR DESCRIPTION
**Pull request by bastilian**
In compliance we used to have a warning in the InventoryTable on specific pages under certain conditions, which we injected via the rows in the reducer.
With a recent refactor we removed this. To bring it back and not again inject it via the reducer, it would be nice to be able and pass a custom No Systems Table for the Entity Table to display.

Is this a good approach or could this be done differently?

See this for a usage example and screenshot of what the empty state with the warning should look like: https://github.com/RedHatInsights/compliance-frontend/pull/1170

### CODE REVIEW

File: packages/inventory/src/components/table/__snapshots__/EntityTable.test.js.snap [hunk 1]
```diff
@@ -4,7 +4,7 @@ exports[`EntityTable DOM should render correctly - compact 1`] = `
 <table
   aria-label="Host inventory"
   class="ins-c-entity-table pf-c-table pf-m-grid-md pf-m-compact"
-  data-ouia-component-id="OUIA-Generated-Table-20"
+  data-ouia-component-id="OUIA-Generated-Table-22"
```
Comment (bastilian) [inline packages/inventory/src/components/table/__snapshots__/EntityTable.test.js.snap:7]: I'm unsure why these change and if this is a valid change. I only pushed the snapshot with the results of the test added here, because I thought it might be a different patternfly version difference or something.

File: packages/inventory/src/components/table/helpers.js [hunk 2]
```diff
@@ -18,11 +18,11 @@ export const buildCells = (item, columns, extra) => {
     });
 };
 
-export const createRows = (rows = [], columns = [], { actions, expandable, ...extra } = {}) => {
+export const createRows = (rows = [], columns = [], { actions, expandable, customNoSystemsTable, ...extra } = {}) => {
     if (rows.length === 0) {
         return [{
             cells: [{
-                title: <NoSystemsTable />,
+                title: customNoSystemsTable ? customNoSystemsTable : <NoSystemsTable />,
```
Comment (rvsia) [inline packages/inventory/src/components/table/helpers.js:25]: The same as in the previous PR, we can skip `custom` from props and make the default component as the default prop.

File: packages/inventory/src/components/table/__snapshots__/EntityTable.test.js.snap (hunk 1 above)
Comment (rvsia) [reply packages/inventory/src/components/table/__snapshots__/EntityTable.test.js.snap:7]: Yeah, this is valid. PF generates ids automatically and deterministically, and because you added new test cases, there are now more components rendered in the same “instance” so the global ID pool is now bigger. If you run this test seperately, the number will be 1.

... and that’s one of the reasons I don’t like snapshot tests. 😄

**Review by rvsia — COMMENTED** [review]
Looks great! 🍹

File: packages/inventory/src/components/table/__snapshots__/EntityTable.test.js.snap (hunk 1 above)
Comment (bastilian) [reply packages/inventory/src/components/table/__snapshots__/EntityTable.test.js.snap:7]: :D oh, ok. yeah, the snapshot diffs can get excessive.

File: packages/inventory/src/components/table/EntityTable.js [hunk 3]
```diff
@@ -139,7 +142,8 @@ EntityTable.propTypes = {
         [PropTypes.string]: PropTypes.any
     }),
     onRowClick: PropTypes.func,
-    showTags: PropTypes.bool
+    showTags: PropTypes.bool,
+    customNoSystemsTable: PropTypes.node
```
Comment (rvsia) [inline packages/inventory/src/components/table/EntityTable.js:146]: Old name

**Review by rvsia — APPROVED** [review]

File: packages/inventory/src/components/table/helpers.test.js [hunk 4]
```diff
@@ -42,15 +42,11 @@ describe('createRows', () => {
     const cells = [{ key: 'first' }, { key: 'second.dot' }, { key: 'third' }];
     it('should create empty table', () => {
         const data = createRows();
-        const wrapper = mount(data[0].cells[0].title);
```
Comment (bastilian) [inline packages/inventory/src/components/table/helpers.test.js:45]: The component is now outside the helpers, the tests still pass and the actual rendering of the `NoSystemsResults` component is tested in the `InventoryTable` tests.

**Review by karelhala — APPROVED** [review]
Looking good!
